src/main/java/com/google/error_analyzer/backend/MockLogDao.java

```diff
@@ -29,7 +30,8 @@
     "WARNING: An illegal reflective access operation has occurred", 
     "Severe: Could not find index file", "warning: NullPointerException"};
     private ArrayList < Index > logDatabase = new ArrayList < Index >();
-
+    private ArrayList<String> errorDatabase;
```

Comment:
I got slightly confused by database, logDatabase & errorDatabase.
More so because one is of type "Index" & other two of type "String".
Try to see if we can reduce the confusion.

Reply:
I'll add comments to clarify that. Changing the names can mess with someone else's code